src/model.js

```diff
@@ -1102,41 +1149,49 @@ ${associationOwner._getAssociationDebugList()}`);
       }
 
       if (Object.prototype.hasOwnProperty.call(definition, 'unique') && definition.unique) {
-        let idxName;
-        if (
-          typeof definition.unique === 'object'
-          && Object.prototype.hasOwnProperty.call(definition.unique, 'name')
-        ) {
-          idxName = definition.unique.name;
-        } else if (typeof definition.unique === 'string') {
-          idxName = definition.unique;
-        } else {
-          idxName = `${this.tableName}_${name}_unique`;
+        if (typeof definition.unique === 'string') {
+          definition.unique = {
+            name: definition.unique,
+          };
+        } else if (definition.unique === true) {
+          definition.unique = {};
         }
 
-        const idx = this.uniqueKeys[idxName] || { fields: [] };
+        const index = definition.unique.name && this.uniqueKeys[definition.unique.name]
+          ? this.uniqueKeys[definition.unique.name]
+          : { fields: [] };
+
+        index.fields.push(definition.field);
+        index.msg = index.msg || definition.unique.msg || null;
+
+        // TODO: remove this 'column'? It does not work with composite indexes, and is only used by db2 which should use fields instead.
```

Comment:
If we remove this, that would be a breaking change right? Might be good to decide on that before releasing v7

Reply:
Maybe. It was added a long time ago (2016 (wow 2016 was a long time ago?)) in this commit https://github.com/sequelize/sequelize/commit/47dc46a7f4b00e06687c0bdea9a53a5cc2ba18fd#diff-305dbcc98ad8fc959dd0c0be51280893962af34c2d8b18abc871cffc2d80c5f9R858

It's possible some people rely on it, so we should consider it to be breaking. Not a huge priority for v7, keeping it doesn't really impact us at all

Reply:
True, let's just leave it in for now until somebody wants to work on this